sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -581,7 +581,6 @@ object FunctionRegistry {
         val params = Seq.fill(expressions.size)(classOf[Expression])
         val f = constructors.find(_.getParameterTypes.toSeq == params).getOrElse {
           val validParametersCount = constructors
-            .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
```

Comment:
This partially reverts #21226. Previous logic was to detect SQL related expressions only but now it includes all constructors that are not related to SQL expressions. Can you elaborate why we revert?

Looks we should rather specially handle `In`'s constructors or handle when `validParametersCount` is empty.

Reply:
SQL related expressions has paras which are not just `Expression`s
This logic was added in #21226, I have no idea why this should add in that pr. Seems not related to its topic.

Reply:
Constructors with expressions only are recognised in SQL.so if a constructor has non-`Expression` argument, it should be excluded.

Reply:
Constructors may take 
```
1 Expression*; 
2 Seq[Expression]; 
3 (Expression, Expression*);
4 (Expression, Seq[Expression]);
```
as args

Reply:
Yea, then can you consider them except other parameters rather than reverting the filter logic entirely?

Reply:
I think we don't support `4 (Expression, Seq[Expression]);` case here.